src/win32/posix_w32.c

```diff
@@ -145,7 +137,7 @@ static bool path_is_volume(wchar_t *target, size_t target_len)
 
 /* On success, returns the length, in characters, of the path stored in dest.
  * On failure, returns a negative value. */
-static int readlink_w(
+int readlink_w(
```

Comment:
Making it non-static means we need to give it a namespaced name, and possibly move it out of this file, as this is only about implementing the posix-like functions. `git_win32_path_readlink_w()` perhaps?